internal/expr/typeinfo.go

```diff
@@ -109,9 +112,59 @@ func getTypeInfo(value any) (typeInfo, error) {
 	return typeInfo, nil
 }
 
+// getStructFields requires the caller to check that t is a struct.
+func getStructFields(t reflect.Type) ([]*structField, error) {
+	var fields []*structField
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		if !f.IsExported() {
+			return nil, fmt.Errorf("field %q of struct %s not exported", f.Name, t.Name())
+		}
+		ft := f.Type
+		k := ft.Kind()
+		// Check if the field type is an embedded/nested struct or pointer to
+		// one. If a pointer to the field type implementes Scanner then it is
+		// not a nested struct.
+		if (k == reflect.Struct && !reflect.PointerTo(ft).Implements(scannerInterface)) ||
+			(k == reflect.Pointer && ft.Elem().Kind() == reflect.Struct && !ft.Implements(scannerInterface)) {
+			if k == reflect.Pointer {
+				ft = ft.Elem()
+			}
+			nestedFields, err := getStructFields(ft)
+			if err != nil {
+				return nil, err
+			}
+			for _, nestedField := range nestedFields {
```

Comment:
Maybe something like this could help others ?

```
// Promote in nested struct fields into the current parent struct scope, but make
// sure we update the Index list to help us navigate back to the original nested
// location for access to the real location.
```

Reply:
Added comment.